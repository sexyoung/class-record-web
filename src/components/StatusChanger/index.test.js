import { render, screen } from '@testing-library/react';
import UserEvent from "@testing-library/user-event";
import { StatusChanger } from '.';

const student = {
  email: "hfjdjd@djdur.djd",
  expiresAt: "2022-02-13 15:04:51",
  id: 28,
  isExpired: false,
  lineId: null,
  name: "WALLET",
  phone: "0930123456",
  picture: "e52dfdb11ca1691e0e266f4e82a428a9.png",
  planTimes: 8,
  rollcalls: 1,
  status: "join",
  times: 7
};

it('should changeStatus be triggered', () => {
  const changeStatus = jest.fn();
  const { container, getByRole } = render(
    <StatusChanger
      status='dropout'
      student={student}
      changeStatus={changeStatus}
    />);
  UserEvent.click(getByRole('button'));
  expect(changeStatus.mock.calls.length).toBe(1);
  expect(container).toMatchSnapshot();
});

