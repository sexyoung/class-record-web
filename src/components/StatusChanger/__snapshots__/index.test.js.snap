// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should changeStatus be triggered 1`] = `
<div>
  <div
    class="StatusChanger"
  >
    <div
      class="title"
    >
      除籍
    </div>
    <div
      class="img"
    />
    <div
      class="name"
    >
      WALLET
    </div>
    <button
      class="btn w-full mt-3"
    >
      除籍
    </button>
  </div>
</div>
`;
